### PR TITLE
Feat/aapd 1112/rule 6

### PR DESCRIPTION
**Pull request by DeanH1990**
## Ticket Number

Rule 6 party appeal page sections
https://pins-ds.atlassian.net/browse/AAPD-1112
https://pins-ds.atlassian.net/browse/AAPD-1114
https://pins-ds.atlassian.net/browse/AAPD-1118
https://pins-ds.atlassian.net/browse/AAPD-1120
https://pins-ds.atlassian.net/browse/AAPD-1123

Rule 6 party model
https://pins-ds.atlassian.net/browse/AAPD-1335

## Description of change

This PR adds the sections to the appeal page for the rule 6 party. 
It differentiates the current rule 6 user from other rule 6 parties to show them their documents/ evidence

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by Si3rr4wow — APPROVED** [review]
Well this turned out to be a lot more work than we made it out to be 😅

File: packages/forms-web-app/src/controllers/selected-appeal/rule-6-sections.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		links: [
+			{
+				url: '/your-statement',
+				text: 'View your statement',
+				condition: (appealCase, userEmail) => {
+					const currentRule6User = appealCase.Rule6Parties?.find(
+						(party) => party.partyEmail === userEmail
+					);
+					return currentRule6User?.statementReceived;
+				}
```
Comment (beejjacobs) [inline packages/forms-web-app/src/controllers/selected-appeal/rule-6-sections.js:39]: This worked out quite nicely :)

**Review by beejjacobs — CHANGES_REQUESTED** [review]
Apologies, some model change suggestions please.

Rest of the code and implementation looks good. The dynamic part is working well.

**Review by beejjacobs — APPROVED** [review]
Good work!

**Pull request merged**